src/rpc/util.cpp

```diff
@@ -41,19 +41,15 @@ void RPCTypeCheck(const UniValue& params,
 
         const UniValue& v = params[i];
         if (!(fAllowNull && v.isNull())) {
-            RPCTypeCheckArgument(v, t);
+            if (!t.typeAny && v.type() != t.type) {
+                throw JSONRPCError(RPC_TYPE_ERROR,
+                                   strprintf("JSON value of type %s is not of expected type %s", uvTypeName(v.type()), uvTypeName(t.type)));
+            }
```

Comment:
maybe merge if statements?

Reply:
I'm unsure about chaining all those statements together. It's hard to read already.

What about splitting the statements?
e.g.
```c++
if (fAllowNull && v.isNull()) continue;
if (!t.typeAny && v.type() != t.type) {
    throw JSONRPCError(RPC_TYPE_ERROR,
          strprintf("JSON value of type %s is not of expected type %s", uvTypeName(v.type()), uvTypeName(t.type)));
}
```

Reply:
looks good :+1: